packages/backend/src/app/flows/flow-version/flow-version.service.ts

```diff
@@ -194,14 +194,51 @@ function getImportOperations(step: Action | Trigger | undefined): (FlowOperation
     }
     return steps
 }
- 
-function keepBaseAction(action: Action): Action{
-    const cloned = JSON.parse(JSON.stringify(action))
-    delete cloned.nextAction
-    delete cloned.onFailureAction
-    delete cloned.onSuccessAction
-    delete cloned.firstLoopAction
-    return cloned
+
+// It's better to use switch case, to enforce that all actions are covered
+function keepBaseAction(action: Action): Action {
+    switch(action.type) {
+        case ActionType.BRANCH:
+            return {
+                type: ActionType.BRANCH,
+                name: action.name,
+                displayName: action.displayName,
+                settings: action.settings,
+                valid: action.valid,
+            }
+        case ActionType.LOOP_ON_ITEMS:
+            return {
+                type: ActionType.LOOP_ON_ITEMS,
+                name: action.name,
+                displayName: action.displayName,
+                settings: action.settings,
+                valid: action.valid,
+            }
+        case ActionType.CODE:
+            return {
+                type: action.type,
+                name: action.name,
+                displayName: action.displayName,
+                settings: action.settings,
+                valid: action.valid,
+            }
+        case ActionType.PIECE:
+            return {
+                type: action.type,
+                name: action.name,
+                displayName: action.displayName,
+                settings: action.settings,
+                valid: action.valid,
+            }
+        case ActionType.MISSING:
+            return {
+                type: action.type,
+                name: action.name,
+                displayName: action.displayName,
+                settings: action.settings,
+                valid: action.valid,
+            }
+    }
```

Comment:
correct me if I'm wrong, but this function can be reduced to this, no?
```ts
function keepBaseAction(action: Action): Action {
    return {
        type: action.type,
        name: action.name,
        displayName: action.displayName,
        settings: action.settings,
        valid: action.valid,
    } as Action
}
```

Reply:
The reason behind this, you won't get compilation error when something is missing.